eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -14,17 +14,32 @@ steps:
     displayName: Create PRData Artifact
 
   - ${{ each repo in parameters.Repos }}:
-    - pwsh: |
-        git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
-        $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
+    - ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
+      - pwsh: |
+          git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
+          $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
+
+          if (!(Test-Path $repoPath)) { mkdir $repoPath }
+          Remove-Item -v -r $repoPath
+          Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
+          Get-ChildItem -r $repoPath
+        displayName: Copy ${{ parameters.DirectoryToSync }} from azure-sdk-tools to ${{ repo }}
+        workingDirectory: $(System.DefaultWorkingDirectory)
 
-        if (!(Test-Path $repoPath)) { mkdir $repoPath }
-        Remove-Item -v -r $repoPath
-        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
-        Get-ChildItem -r $repoPath
+    - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+      - pwsh: |
+          git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
+          $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
+          $patchDir = "${{ repo }}/patchfiles"
 
-      displayName: Copy ${{ parameters.DirectoryToSync }} from azure-sdk-tools to ${{ repo }}
-      workingDirectory: $(System.DefaultWorkingDirectory)
+          pushd $(Build.SourcesDirectory)
+          git format-patch ${{ parameters.BaseBranchName }}...upstream/master ${{ parameters.DirectoryToSync }} -o $patchDir
+          popd
+          pushd ${{ repo }}
+          Get-ChildItem $patchDir | % { git am $_.FullName }
```

Comment:
We need to makes sure we handle failures correctly here.

Reply:
git am will actually make commits but the other parts of this yml file below assume there is a dirty repo and it will try to add and make a commit and if there is no diff it will skip the push and pull request creation. We might need to make some changes to account for this new scenario.